mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/KeyValueBasedPersistenceProvider.java

```diff
@@ -421,23 +421,23 @@ public List<IJobClusterMetadata> loadAllJobClusters() throws IOException {
     public List<CompletedJob> loadAllCompletedJobs() throws IOException {
         AtomicInteger failedCount = new AtomicInteger();
         AtomicInteger successCount = new AtomicInteger();
-
-        final List<CompletedJob> completedJobsList = Lists.newArrayList();
-        final String namespace = NAMED_COMPLETEDJOBS_NS;
-        for (String pkey : kvStore.getAllPartitionKeys(namespace)) {
-            kvStore.getAll(namespace, pkey).values()
-                .forEach(data -> {
-                    try {
-                        NamedJob.CompletedJob cj = mapper.readValue(data, NamedJob.CompletedJob.class);
-                        CompletedJob completedJob = DataFormatAdapter.convertNamedJobCompletedJobToCompletedJob(cj);
-                        completedJobsList.add(completedJob);
-                        successCount.getAndIncrement();
-                    } catch (JsonProcessingException e) {
-                        logger.warn("failed to parse CompletedJob from {} {}", pkey, data, e);
-                        failedCount.getAndIncrement();
-                    }
-                });
-        }
+        final List<CompletedJob> completedJobsList = kvStore.getAllRows(NAMED_COMPLETEDJOBS_NS)
```

Comment:
I noticed that `getAllRows` still uses `getAllPartitionKeys`. is there any major performance difference between the old and new code? Also is the docstring on `getAllPartitionKeys` still valid? https://github.com/Netflix/mantis/blob/03685037d1a3903068054dcf4d9526e6a2fb8b4b/mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/store/KeyValueStore.java#L56

Reply:
That's the default implementation but the inheritors can choose to override the implementation. 
Also, this change wouldn't make this function call fast enough, with the current implementation and table design. I'm disabling it since the older implementation wasn't doing anything either.